Replace `simplelog` with `tracing` and `tracing-subscriber`

Kajmany opened the pull request:
As requested in #33 
I tried to (roughly) replicate the current logging output using the `fmt` subscriber. Lots of stuff can be twiddled with format and information though. My thought on the new way to set log levels is that log levels below INFO on deps are probably only useful during development  (where setting `RUST_LOG` is not a big deal) - so I use the current verbosity flag to just change this crate's log level.

Other ideas:
- More layers can be added to expose events to outside *things* like Tokio console, outside log aggregators, etc.
- Spans can be used to give more useful context to work units. [The macro](https://docs.rs/tracing/latest/tracing/attr.instrument.html) is recommended to avoid footguns.

Kajmany committed:
Replace simplelog with tracing for emission (drop-in!) and tracing-subscriber for reading

Kajmany committed:
Remove simplelog, update clap-verbosity-flag

detjensrobert commented on src/main.rs:
```diff
@@ -1,23 +1,26 @@
-    )
-    .unwrap();
+    // Use RUST_LOG env variable if it's set
+    // Otherwise our tracing is filtered by -q|-v* flag, all others always INFO and above
```
How hard would it be to set others to TRACE/DEBUG with `-vvv`/`-vvvv` instead of/in addition to RUST_LOG? since we're starting at INFO instead of silent I'm not sure if the clab_verbosity_flag will go past `-vv` here.

Kajmany replied:
'instead of' is not hard at all - I can abuse `clap_verbosity_flag`'s enums horribly, or put in only slightly more effort to lift their relevant code (it's a tiny lib) and drop the dependency.

'in addition to' depends on what you have in mind for combining the two. I believe it's traditional that `RUST_LOG` can override the log-levels for everything, but there's nothing stopping us from having it set dependency log levels while `-v` sets ours. Allowing CLI and env to influence both is messier - I can't think of a totally general way to do it.

detjensrobert replied:
Having RUST_LOG set it for everything is fine -- if the user is setting that we can safely assume they want the whole spew. 

Creating our own -v flag for this seems pretty simple, lets do that.

detjensrobert reviewed (changes requested):
Couple things:
- If we can, I'd like to keep INFO without the span information -- thats a little too verbose for the default user-facing messages. 
- Simplelog allowed for colors in log messages, like https://github.com/osusec/beavercds-ng/blob/2ca0f4af5fffa5d2a1f1e97e75a84baff1755c20/src/cluster_setup/mod.rs#L164. Is that something we can do with `tracing` still or do we need to pull in `paris`/similar for that?

Kajmany committed:
Make log-level setting more granular. Hide spans at most levels. Demo simple timestamp.

Core functionality of counting -v flags was taken from `clap_verbosity_flag`, but we match our own custom verbosity now to allow setting different log levels and other settings.

Spans are hidden on the first two levels of verbosity.

Pre-existing Chrono dep is used to create a more compact timestamp in logs with strftime-like formatting.

Kajmany committed:
convert `paris` coloring to `owo-colors` which is more compatible with tracing

Kajmany committed:
Get a lot more info (nanoseconds, span in/out, thread id) at -vv+ verbosity levels

detjensrobert reviewed (approved):
Looks good! I'm happy with how this works. The crash has been fixed in the latest main if you rebase :)
Once you rebase, I'll merge this in.

I'm eh on having timestamps for INFO level, but I won't block merging this on that.

Kajmany committed:
Migrate final module to tracing. Return vandalized dependency
